specs/eventing/data-plane.md

```diff
@@ -1,125 +1,179 @@
-`405 Method Not Supported`. Non-event requests (e.g. health checks) are not
-constrained.
-
-The URL used by a Sink MUST correspond to a single, unique endpoint at any given
-moment in time. This MAY be done via the host, path, query string, or any
-combination of these. This mapping is handled exclusively by the
-[Addressable control-plane](./interfaces.md#control-plane) exposed via the
-`status.address.url`.
-
-If an HTTP request's URL does not correspond to an existing endpoint, then the
-Sink MUST respond with `404 Not Found`.
-
-Every non-Callable Sink MUST respond with `202 Accepted` if the request is
-accepted.
-
-If Sink is Callable it MAY respond with `200 OK` and a single event in the HTTP
-response. A returned event is not required to be related to the received event.
-The Callable should return a successful response if the event was processed
-successfully. If there is no event to send back then Callable Sink MUST respond
-with 2xx HTTP and with empty body.
-
-If a Sink receives a request and is unable to parse a valid CloudEvent, then it
-MUST respond with `400 Bad Request`.
-
-### Content Modes Supported
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in RFC2119.
 
-A Sink MUST support `Binary Content Mode` and `Structured Content Mode` as
-described in
-[HTTP Message Mapping section of HTTP Protocol Binding for CloudEvents](https://github.com/cloudevents/spec/blob/master/http-protocol-binding.md#3-http-message-mapping)
+When not specified in this document, the
+[CloudEvents HTTP bindings, version 1.0](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md)
+and [HTTP 1.1 protocol](https://tools.ietf.org/html/rfc7230) standards should be
+followed (with the CloudEvents bindings preferred in the case of conflict).
 
-A Sink MAY support `Batched Content Mode` but that mode is not used in Knative
-Eventing currently (that may change in future).
+The current version of this document does not describe protocol negotiation or
+any delivery mechanism other than HTTP 1.1. Future versions may define protocol
+negotiation to optimize delivery; compliant implementations SHOULD aim to
+interoperate by ignoring unrecognized negotiation options (such as
+[HTTP `Upgrade` headers](https://datatracker.ietf.org/doc/html/rfc7230#section-6.7)).
 
-### Retries
+## Event Delivery
 
-Sinks should expect that retries and accept possibility that duplicate events
-may be delivered.
+To provide simpler support for event sources which might be translating events
+from existing systems, some data plane requirements for senders are relaxed in
+the general case. In the case of Knative Eventing provided resources (Channels
+and Brokers) which implement these roles, requirements are increased from SHOULD
+to MUST. Exceptions with stronger requirements for Channels and Brokers are
+called out as they occur.
```

Comment:
I think I understand what this paragraph is saying, but what's not clear to me is it it's just an "FYI" thing or whether this is implying that the reader should mentally map some requirements down for "existing systems" and then up (from SHOULD to MUST) for brokers/channels? If this is really just an FYI thing to explain to people why the general requirements are SHOULDs but then Broker/Channels might turn some into MUSTs, I'd almost prefer to just remove this paragraph entirely for fear it 1) adds little value, 2) it might confuse people as to what they need to do about it while coding.... which is "nothing".

Reply:
(Removed this paragraph)